From c2825413158765b272899f892affa9dd87d94d30 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 31 Dec 2017 14:22:36 +0200
Subject: [PATCH 30/57] nxp: pcie: External OSC handling

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/pci/host/pci-imx6.c | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index cb4cf08..e0dbc54 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -79,6 +79,7 @@ struct imx6_pcie {
 	struct clk		*pcie;
 	struct clk		*pcie_ext_src;
 	struct regmap		*iomuxc_gpr;
+	struct regmap		*ccm_base;
 	enum imx6_pcie_variants variant;
 	u32			hsio_cfg;
 	u32			tx_deemph_gen1;
@@ -1064,9 +1065,16 @@ static void imx6_pcie_init_phy(struct imx6_pcie *imx6_pcie)
 		else
 			val = IOMUXC_GPR16;
 
-		regmap_update_bits(imx6_pcie->iomuxc_gpr, val,
-				IMX8MQ_GPR_PCIE_REF_USE_PAD,
-				IMX8MQ_GPR_PCIE_REF_USE_PAD);
+		if (imx6_pcie->ext_osc) {
+			regmap_update_bits(imx6_pcie->iomuxc_gpr, val,
+					IMX8MQ_GPR_PCIE_REF_USE_PAD,
+					IMX8MQ_GPR_PCIE_REF_USE_PAD);
+		} else {
+			regmap_update_bits(imx6_pcie->iomuxc_gpr, val,
+					IMX8MQ_GPR_PCIE_REF_USE_PAD,
+					0);
+		}
+
 	} else if (imx6_pcie->variant == IMX7D) {
 		/* Enable PCIe PHY 1P0D */
 		regulator_set_voltage(imx6_pcie->pcie_phy_regulator,
@@ -2219,6 +2227,13 @@ static int imx6_pcie_probe(struct platform_device *pdev)
 				"imx8mq pcie phy src missing or invalid\n");
 			return PTR_ERR(imx6_pcie->reg_gpc);
 		}
+
+		if (!imx6_pcie->ext_osc) {
+			imx6_pcie->ccm_base=syscon_regmap_lookup_by_compatible("fsl,imx8mq-anatop");
+			regmap_update_bits(imx6_pcie->ccm_base, 0x74, 0xFFFFFFFF, 0x1B);
+			regmap_update_bits(imx6_pcie->ccm_base, 0x7C, 0xFFFFFFFF, 0x77777);
+		}
+
 	} else if (imx6_pcie->variant == IMX6SX) {
 		imx6_pcie->pcie_inbound_axi = devm_clk_get(&pdev->dev,
 				"pcie_inbound_axi");
-- 
1.9.1

